testsuite/shared/src/main/java/org/jboss/as/test/config/ContainerConfig.java

```diff
@@ -0,0 +1,295 @@
+     *
+     * @return the OpenTelemetry Collector image name
+     */
+    public static String getOpenTelemetryCollectorImageName() {
+        return getOpenTelemetryCollectorImage().split(VERSION_SEPARATOR)[0];
+    }
+
+    /**
+     * Returns the version portion of the OpenTelemetry Collector container image (after the colon).
+     *
+     * @return the OpenTelemetry Collector image version
+     */
+    public static String getOpenTelemetryCollectorImageVersion() {
+        return getOpenTelemetryCollectorImage().split(VERSION_SEPARATOR)[1];
+    }
+
+    /*
+        Elasticsearch
+     */
+
+    /**
+     * Returns the full Elasticsearch container image reference (name:version).
+     *
+     * @return the Elasticsearch image reference
+     */
+    public static String getElasticsearchImage() {
+        return getProperty(ELASTICSEARCH_IMAGE_CONFIG_KEY, ELASTICSEARCH_DEFAULT_IMAGE);
+    }
+
+    /*
+        MailServer
+     */
+
+    /**
+     * Returns the full Mail Server container image reference (name:version).
+     *
+     * @return the Mail Server image reference
+     */
+    public static String getMailServerImage() {
+        return getProperty(MAILSERVER_IMAGE_CONFIG_KEY, MAILSERVER_DEFAULT_IMAGE);
+    }
+
+    /*
+        Kafka
+     */
+
+    /**
+     * Returns the full Kafka container image reference (name:version).
+     *
+     * @return the Kafka image reference
+     */
+    public static String getKafkaImage() {
+        return getProperty(KAFKA_IMAGE_CONFIG_KEY, KAFKA_DEFAULT_IMAGE);
+    }
+
+    /**
+     * Returns the name portion of the Kafka container image (before the colon).
+     *
+     * @return the Kafka image name
+     */
+    public static String getKafkaImageName() {
+        return getKafkaImage().split(VERSION_SEPARATOR)[0];
+    }
+
+    /**
+     * Returns the version portion of the Kafka container image (after the colon).
+     *
+     * @return the Kafka image version
+     */
+    public static String getKafkaImageVersion() {
+        return getKafkaImage().split(VERSION_SEPARATOR)[1];
+    }
+
+    /*
+        Artemis
+     */
+
+    /**
+     * Returns the full Artemis container image reference (name:version).
+     *
+     * @return the Artemis image reference
+     */
+    public static String getArtemisImage() {
+        return getProperty(ARTEMIS_IMAGE_CONFIG_KEY, ARTEMIS_DEFAULT_IMAGE);
+    }
+
+    /*
+        Keycloak
+     */
+
+    /**
+     * Returns the full Keycloak container image reference (name:version).
+     * <p>
+     * Note: This method also checks the legacy property 'testsuite.integration.oidc.rhsso.image'
+     * for backwards compatibility.
+     *
+     * @return the Keycloak image reference
+     */
+    public static String getKeycloakImage() {
+        return System.getProperty("testsuite.integration.oidc.rhsso.image", getProperty(KEYCLOAK_IMAGE_CONFIG_KEY, KEYCLOAK_DEFAULT_IMAGE));
+    }
+}
```

Comment:
All of these specific methods seem unnecessary and this whole class should just be an `enum`.

Then you can simply just call e.g. `ContainerConfig.OTEL_COLLECTOR.getImageName()`

Reply:
Changes applied, resolving conversation.